src/main/scala/cromwell/engine/WorkflowDescriptor.scala

```diff
@@ -65,6 +73,51 @@ case class WorkflowDescriptor(id: WorkflowId,
     case _ => NOPLogger.NOP_LOGGER
   }
 
+  /**
+    * Delete authentication file in GCS once workflow is in a terminal state.
+    *
+    * First queries for the existence of the auth file, then deletes it if it exists.
+    * If either of these operations fails, then a Future.failure is returned
+    */
+  def postProcessWorkflow(implicit executionContext: ExecutionContext): Future[Any] = {
+    // Try to copy outputs to final destination
+    workflowOptions.get(CopyOutputsOptionKey) map copyOutputFiles getOrElse Future.successful({})
+  }
+
+  private def copyOutputFiles(destDirectory: String)(implicit executionContext: ExecutionContext): Future[Unit] = {
+    import cromwell.util.PathUtil._
+    val logger = backend.workflowLogger(this)
+    implicit val gcsFs = gcsFilesystem
+
+    def copyFile(file: WdlFile) = {
+      val src = file.valueString.toPath
+      val wfPath = wfContext.root.toPath.toAbsolutePath // Absolute path of the workflow directory
+      val relativeFilePath = src.subpath(wfPath.getNameCount, src.getNameCount) // Path of the src file relatively to the workflow directory
+      val dest = destDirectory.toPath.resolve(relativeWorkflowRootPath).resolve(relativeFilePath) // Path of destination file
+
+      def copy = {
+        logger.info(s"Trying to copy output file $src to $dest")
+        Files.createDirectories(dest.getParent)
+        Files.copy(src, dest)
+      }
+
+      TryUtil.defaultRetry(copy, logger, s"Failed to copy file $src to $dest", Seq(classOf[FileAlreadyExistsException])) recover {
+        case alreadyExists: FileAlreadyExistsException => logger.info(s"Tried to copy the same file multiple times. Skipping subsequent copies for $src")
```

Comment:
Oh I wasn't suggesting this should fail, just that it might want to log the exception.  But I guess that's not actually going to give any new information; if $src can't be copied because there's already a file at $src, that's really all there is to say.  :smile: